workflow/installertest.groovy

```diff
@@ -0,0 +1,133 @@
+// Replace colons in image with hyphens and append text
+String convertImageNameToString(String imageName, String append="") {
+    return (imageName+append).replaceAll(':','-')
+}
+
+/**
+ * Extracts the components from an {@code 'artifact://full/path/to/job/buildNr#artifact.ext'} type url.
+ * @param url the url
+ */
+@NonCPS
+def getComponentsFromArtifactUrl(String url) {
+    def pattern = /^artifact:\/\/([\/\w-_ ]+)\/(\d+)\/{0,1}#([\/\w\.]+)$/
```

Comment:
My suggestion, from experiments - `def pattern = /^artifact:\/\/([\/\w\-\_\. ]+)\/(\d+)\/{0,1}#([\/\w\.\-\_]+)$/`. Weirdly, without escaping `-`, if it was following `\.`, it failed to match correctly, while `def pattern = /^artifact:\/\/([\/\w-_\. ]+)\/(\d+)\/{0,1}#([\/\w-_\.]+)$/` did fine. So yeah, your pick?